src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -40,130 +41,118 @@
+    /**
+     * Spark HTTP endpoint to get a single feed source by ID.
+     */
+    private static FeedSource getFeedSource(Request req, Response res) {
         return requestFeedSourceById(req, Actions.VIEW);
     }
 
-    public static Collection<FeedSource> getAllFeedSources(Request req, Response res) {
+    /**
+     * Spark HTTP endpoint that handles getting all feed sources for a handful of use cases:
+     * - for a single project (if projectId query param provided)
+     * - for the entire application
+     */
+    private static Collection<FeedSource> getAllFeedSources(Request req, Response res) {
         Collection<FeedSource> feedSourcesToReturn = new ArrayList<>();
         Auth0UserProfile requestingUser = req.attribute("user");
         String projectId = req.queryParams("projectId");
-        Boolean publicFilter = req.pathInfo().contains("public");
-        String userId = req.queryParams("userId");
-
-        if (projectId != null) {
-            for (FeedSource source: Persistence.feedSources.getAll()) {
-                String orgId = source.organizationId();
-                if (
-                    source != null && source.projectId != null && source.projectId.equals(projectId)
-                    && requestingUser != null && (requestingUser.canManageFeed(orgId, source.projectId, source.id) || requestingUser.canViewFeed(orgId, source.projectId, source.id))
-                ) {
-                    // if requesting public sources and source is not public; skip source
-                    if (publicFilter && !source.isPublic)
-                        continue;
-                    feedSourcesToReturn.add(source);
-                }
-            }
-        } else if (userId != null) {
-            // request feed sources a specified user has permissions for
-            Auth0UserProfile user = getUserById(userId);
-            if (user == null) return feedSourcesToReturn;
-
-            for (FeedSource source: Persistence.feedSources.getAll()) {
+        // Determine whether request is coming from a public HTTP endpoint.
+        boolean publicFilter = req.pathInfo().contains("public");
+        List<FeedSource> allFeedSources = Persistence.feedSources.getAll();
+        if (projectId != null && requestingUser != null) {
+            for (FeedSource source: allFeedSources) {
                 String orgId = source.organizationId();
                 if (
-                    source != null && source.projectId != null &&
-                    (user.canManageFeed(orgId, source.projectId, source.id) || user.canViewFeed(orgId, source.projectId, source.id))
+                    source.projectId != null && source.projectId.equals(projectId) &&
+                    requestingUser.canManageOrViewFeed(orgId, source.projectId, source.id)
                 ) {
-
+                    // If requesting public sources and source is not public; skip source
+                    // TODO: determine if this is still needed.
```

Comment:
Maybe we should determine this now. Didn't we already get rid of a bunch of public endpoints?

Reply:
Yea good point. I'll do a double check, but then just go ahead and remove.